### PR TITLE
feat: get checkpoint endpoint [DET-3734]

### PR DESCRIPTION
**Pull request by sidneyw**
## Description
Singular get checkpoint endpoint. Accepts the uuid of the checkpoint and returns the entire checkpoint.

## Test Plan
Manually tested the endpoint.

## Commentary (optional)
In the future we likely want checkpoint export methods to reference the new API endpoint.

### CODE REVIEW

File: proto/src/determined/api/v1/checkpoint.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// Get the requested checkpoint.
+message GetCheckpointRequest {
+  // The uuid for the requested checkpoint.
+  string checkpoint_uuid = 1;
+}
+
+// Response to GetCheckpointRequest.
+message GetCheckpointResponse {
+  // The requested checkpoint.
+  determined.experiment.v1.Checkpoint checkpoint = 1;
```
Comment (stoksc) [inline proto/src/determined/api/v1/checkpoint.proto:17]: just a question, do you think it makes more sense to have `determined.experiment.v1.Checkpoint checkpoint` live here and be imported in `experiment.proto` or the way it is now?

File: proto/src/determined/experiment/v1/checkpoint.proto [hunk 2]
```diff
@@ -16,6 +17,11 @@ message Metrics {
 
 // Checkpoint is an artifact created by a trial during training.
 message Checkpoint {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+      json_schema: {
+          required: ["uuid", "state"]
+      }
+  };
```
Comment (stoksc) [inline proto/src/determined/experiment/v1/checkpoint.proto:24]: This is neat, wasn't aware of this.

**Comment by sidneyw** [comment]
@stoksc I can change the `determined.experimental.Determined.get_checkpoint` method to reference this endpoint. That would add coverage via integration tests.

File: proto/src/determined/api/v1/checkpoint.proto (hunk 1 above)
Comment (sidneyw) [reply proto/src/determined/api/v1/checkpoint.proto]: Done

File: common/determined_common/experimental/determined.py [hunk 3]
```diff
@@ -44,8 +44,8 @@ def get_checkpoint(self, uuid: str) -> Checkpoint:
         Get the :class:`~determined.experimental.Checkpoint` representing the
         checkpoint with the provided UUID.
         """
-        r = api.get(self._session._master, "checkpoints/{}".format(uuid)).json()
-        return Checkpoint.from_json(r, master=self._session._master)
+        r = api.get(self._session._master, "/api/v1/checkpoints/{}".format(uuid)).json()
+        return Checkpoint.from_json(r["checkpoint"], master=self._session._master)
```
Comment (stoksc) [inline common/determined_common/experimental/determined.py:48]: This is great, so long as you don't see any issues with it (I'm unfamiliar with these APIs).

**Pull request merged**